redisvl/index/index.py

```diff
@@ -32,6 +32,38 @@
 logger = get_logger(__name__)
 
 
+def _handle_dialect_3(result: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Handle dialect 3 responses by converting JSON-encoded list values to strings.
+
+    Each JSON-encoded string in the result that is a list will be converted:
+    - If the list has one item, it is unpacked.
+    - If the list has multiple items, they are joined into a single comma-separated string.
```

Comment:
It seems like this is the desired behavior from @tylerhutcherson but I feel like I'd rather have a list than a comma-separated string in this situation. I may be missing some deeper context though

Reply:
I think I like that better too!